lib/utils/extract-capturing-group-references.ts

```diff
@@ -136,6 +136,51 @@ type ExtractCapturingGroupReferencesContext = {
     isString: (node: Expression) => boolean
 }
 
+type ArrayMethodName = Exclude<keyof unknown[], "length" | symbol | number>
+const WILL_KNOWN_ARRAY_METHODS: {
+    [key in ArrayMethodName]: {
+        iteration?: number[]
+        result?: "element" | "array" | "iterator"
+    }
+} = {
+    toString: {},
+    toLocaleString: {},
+    pop: { result: "element" },
+    push: {},
+    concat: { result: "array" },
+    join: {},
+    reverse: { result: "array" },
+    shift: { result: "element" },
+    slice: { result: "array" },
+    sort: { iteration: [0, 1], result: "array" },
```

Comment:
`iterator` should probably be renamed. It really is a list of all parameter indexes that will be assigned an array element (if the first argument of the array method is a function).

Maybe `elementParams` or `elementParameters`?

Also, could you please document this on the type? It took me a little while to figure this out.